BlockEntity/BEFruitpress.cs

```diff
@@ -263,29 +276,17 @@ private void onTick100msServer(float dt)
                 }
 
                 juiceableLitresLeft -= actuallyTransfered;
+                squeezedLitresLeft -= pressSqueezeRel <= squeezeRel ? actuallyTransfered : (actuallyTransfered * 20); // Let the mash drain less if screw is released.
                 juiceableLitresTransfered += actuallyTransfered;
                 lastLiquidTransferTotalHours = totalHours;
                 MarkDirty(true);
             }
-
-
-            if (juiceableLitresLeft <= 0.01)
+            else if (Api.Side == EnumAppSide.Server && (!CompressAnimActive || juiceableLitresLeft <= 0))
             {
-                // Hack to fix rounding errors
-                if (cntBlock != null)
-                {
-                    float litres = cntBlock.GetCurrentLitres(BucketSlot.Itemstack);
-                    cntBlock.SetCurrentLitres(BucketSlot.Itemstack, (float)Math.Round(100 * litres) / 100f);
-                }
-
+                squeezedJuiceLeft = false;
                 UnregisterGameTickListener(listenerId);
                 listenerId = 0;
 
-                int stacksize = GameMath.RoundRandom(Api.World.Rand, (float)juiceableLitresTransfered);
-                mashStack.Attributes.RemoveAttribute("juiceableLitresTransfered");
-                mashStack.Attributes.RemoveAttribute("juiceableLitresLeft");
-                mashStack.StackSize = (int)(stacksize * juiceProps.PressedDryRatio);
-

```

Comment:
This is moved into a function that is run when the player removes the mash from the fruit press or the press is broken. That makes it so that players can’t accidentally overwrite the dry mash inside with wet mash by adding more fruit, keeping the behavior of mash inside the machine more predictable.